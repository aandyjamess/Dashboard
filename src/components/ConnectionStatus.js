import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const ConnectionStatus = ({connected}) => (
  <span className={classnames("connection-status", {connected})}>
    <span className="connection-status-text pull-right">
      Connected
    </span>
    <span className="icon icon-connect-on pull-right" />
  </span>
);

ConnectionStatus.PropTypes = {
  connected: PropTypes.bool
};

ConnectionStatus.defaultProps = {
  connected: false
};

export default ConnectionStatus;
