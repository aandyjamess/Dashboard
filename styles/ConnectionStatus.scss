@import "./variables";

.connection-status {
  display: flex;
  align-items: center;
  padding: 0 15px;
  margin-left: auto;
  margin-right: 24px;
  height: 100%;

  &.connected {
    background-color: $color-1a;
  }

  &-text {
    margin-right: 15px;
  }
}
