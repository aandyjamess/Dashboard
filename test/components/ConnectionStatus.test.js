import React from "react";
import ConnectionStatus from "../../src/components/ConnectionStatus";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  expect(renderer.create(
    <ConnectionStatus connected={false} />
  ).toJSON()).toMatchSnapshot();
});
