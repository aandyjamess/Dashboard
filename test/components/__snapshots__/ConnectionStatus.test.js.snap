// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="connection-status"
>
  <span
    className="connection-status-text pull-right"
  >
    Connected
  </span>
  <span
    className="icon icon-connect-on pull-right"
  />
</span>
`;
